src/com/magento/idea/magento2plugin/actions/generation/dialog/NewEntityDialog.java

```diff
@@ -1362,4 +1374,29 @@ private String getTableEngine() {
 
         return menuReferencesList;
     }
+
+    /**
+     * Autocomplete entity name dependent fields.
+     */
+    private void autoCompleteIdentifiers() {
+        if (getEntityName().isEmpty()) {
+            return;
+        }
+        final String entityName = CamelCaseToSnakeCase.getInstance().convert(getEntityName());
+        final String entityNameLabel = Arrays.stream(entityName.split("_")).map(
+                string -> string.substring(0, 1).toUpperCase(Locale.getDefault())
+                        + string.substring(1)
+        ).collect(Collectors.joining(" "));
+
+        dbTableName.setText(entityName);
+        entityId.setText(entityName.concat("_id"));
+        route.setText(entityName);
+        formLabel.setText(entityNameLabel.concat(" Form"));
+        formName.setText(entityName.concat("_form"));
+        gridName.setText(entityName.concat("_listing"));
```

Comment:
I'm currently checking this, and my question is: What if you don't check the `Create Admin Ui Components` checkbox? Per my understanding, we should completely hide the `Admin Ui Components` tab if we have it unchecked and no other action should be done in scope of Ui Components tab.

Reply:
If you don't check the Create Admin Ui Components checkbox you won't get generated any files related to UI components.

We cannot depend on it during inputs autocompletion. What if we will fill all inputs on the first tab and only then check the Create Admin Ui Components checkbox? That is why better to leave this in this way - it is more clear for the user.

What about hiding the Admin Ui Components tab, it is a good idea. I'll implement it. Thank you!

After it will be implemented, I'll leave a comment.

Reply:
Hiding that tab would be more logical, and it won't confuse the end-user, since he unchecked the checkbox, but still sees the Tab.